### ISSUE TITLE
send() from pointerEvent method seems to corrupt message buffer

### ISSUE THREAD

**Issue by hergenhahn** [novnc/noVNC#874]
I've had a hard to get to work novnc with x11vnc server, avoiding websockify.
After compiling libvncserver from latest sources, I coul connect and got a creen image, but whenever I moved the mouse, the connection broke down. Same in firefox 41.0.1 and IE 11.0.9600.18697.

x11vnc complained about "unknown message type <nr>", where nr seemed to change at random when trying several times. I checked how mouse moves are transmitted and found function pointerEvent in line 1546 of file rfb.js. It teminates with sock.flush(). If I comment out this sock.flush(), novnc works well for me in firefox, while mouse cursor updates in IE are somewhat slow or "jumpy".
I guess, pointerEvent() interrupts another message building process, inserting its message into the patially filled sock._sQ...

**Comment by DirectXMan12** [comment]
Which version of noVNC are you using?

removing sock.flush shouldn't solve an "interruption", because you'd still have have the pointerEvent message being inserted in the middle of the sQ...

**Comment by hergenhahn** [comment]
I downloaded a tarball from page novnc.com. The file is named novnc-noVNC-v0.6.1-425-g74a29f3.tar.gz.
@DirectXMan12: Yes, you are right. But I can only tell you that avoiding the sock.flush() fixes my problem.

**Comment by DirectXMan12** [comment]
that version of noVNC is a bit old at this point.  Can you try with the master branch?

**Comment by CendioOssman** [comment]
He's just one commit behind, so I don't consider that particularly old. ;)

I'm afraid I'm unable to reproduce the issue though. How was your x11vnc started?

**Comment by dcommander** [comment]
I have unfortunately run into the same issue with the latest noVNC code and the WebSockets interface in LibVNCServer.  It can be reproduced using the "blooptest" example in LibVNCServer.

**Comment by dcommander** [comment]
One more data point:  the issue seems to only occur with ws (unencrypted), not with wss.

**Comment by bk138** [comment]
> One more data point: the issue seems to only occur with ws (unencrypted), not with wss.

slightly offtopic: @dcommander would you be so kind and share your wss:// setup so I can include a step-by-step setup howto in LibVNCServer's README? Merci!

**Comment by dcommander** [comment]
There isn't much to it.  You just need to have a valid X.509 certificate, combine the certificate and private key into a single PEM file, and pass that file to utils/launch.sh.  TurboVNC 3.0, which is still in development, automates that somewhat by introducing an option to vncserver that causes it to launch and track a noVNC proxy process along with the Xvnc process, shutting down both when the Xvnc process is shut down.

**Comment by itsjustbrian** [comment]
Running into the same issue with latest noVNC and x11vnc/libvncserver. Crashes on mouse inputs with `unknown message type` logged in x11vnc. Don't have any new insights to add, but this should probably be reopened right? Unless it's a libvncserver issue.

**Comment by dcommander** [comment]
@itsjustbrian It looks like the PR for that issue (https://github.com/LibVNC/libvncserver/pull/300) was closed without being integrated.  Do you see the commit somewhere in the LibVNCServer repository?

**Comment by itsjustbrian** [comment]
@dcommander The same person opened another identical PR (LibVNC/libvncserver#301). Probably just made a mistake on the first one. BTW I got an unencrypted connection working by just proxying x11vnc with websockify. The issue only exists with the direct websocket connection so this is a valid workaround until it's officially fixed.

**Comment by dcommander** [comment]
@itsjustbrian Thanks for the link.  I subscribed to the new PR.  I think this will also fix the issues in the implementation of WS that I attempted to add to the TurboVNC Server (https://github.com/TurboVNC/turbovnc/issues/127), thus allowing that feature to move forward.

**Comment by dcommander** [comment]
@itsjustbrian The feature is now available in the TurboVNC dev/3.0 evolving branch and pre-release builds.  It works for me with the latest noVNC code and with or without encryption (encryption requires specifying the X.509 certificate and private key using the `-x509key` and `-x509cert` arguments to Xvnc.)

**Comment by alechirsch** [comment]
I am also running into this issue with the latest  noVNC `1.2.0` and x11vnc/libvncserver `0.9.16`. I am using non secure web sockets. It seems to connect and then disconnect as soon as I move the cursor a bit. x11vnc spits out messages like this:
```
rfbProcessClientNormalMessage: unknown message type 95
... closing connection
```
